### PR TITLE
OUT-715 Unnecessary scrollbar is shown on task details page when task title is long

### PR DESCRIPTION
**Pull request by arpandhakal**
### Tasks

- [Unnecessary scrollbar is shown on task details page when task title is long](https://linear.app/copilotplatforms/issue/OUT-715/unnecessary-scrollbar-is-shown-on-task-details-page-when-task-title-is)

### Changes

- [x] applied min height of task details editor based on screen sizes, screen width. 

### Testing 

- [LOOM](https://www.loom.com/share/7c8f65777a054c8b8a4cc46a18593dcc)

### Notes 

- Since attachments are on  the way in m2 milestone, kept the min height of the task details editor a bit concised

### CODE REVIEW

File: src/app/globals.css [hunk 1]
```diff
@@ -5,9 +5,28 @@
 }
 
 .tapwrite-details-page {
-  min-height: 75vh;
+  height: calc(100vh - 50vh);
```
Comment (priosshrsth) [inline src/app/globals.css:8]: Can't it be just 50vh instead of using calc?

**Review by rrojan — CHANGES_REQUESTED** [review]
I can still reproduce this issue on tablet devices. It seems like a horizontal scrollbar appears for them as well. 

https://www.loom.com/share/6f25d50af81f4408a0a2844babfee263

**Review by rrojan — APPROVED** [review]
LGTM :trophy:

File: src/app/detail/ui/Sidebar.tsx [hunk 2]
```diff
@@ -79,7 +79,7 @@ export const Sidebar = ({
     }
   }, [tasks, workflowStates])
 
-  const matches = useMediaQuery('(max-width:600px)')
+  const matches = useMediaQuery('(max-width:700px)')
   if (!tasks) return null
```
Comment (aatbip) [inline src/app/detail/ui/Sidebar.tsx:84]: The initial assumption for the medium screen width is 600px which was also shared in this issue and was decided accordingly - https://linear.app/copilotplatforms/issue/OUT-1/create-tokens

Changing this to 700px over here is going to change the overall responsive idea of the entire app.

**Review by aatbip — CHANGES_REQUESTED** [review]

File: src/app/detail/ui/ToggleController.tsx [hunk 3]
```diff
@@ -8,9 +8,9 @@ import { useSelector } from 'react-redux'
 
 export const ToggleController = ({ children }: { children: ReactNode }) => {
   const { showSidebar } = useSelector(selectTaskDetails)
-  const matches = useMediaQuery('(max-width:600px)')
+  const matches = useMediaQuery('(max-width:700px)')
 
-  const nonMobile = useMediaQuery('(min-width:600px)')
+  const nonMobile = useMediaQuery('(min-width:700px)')
```
Comment (aatbip) [inline src/app/detail/ui/ToggleController.tsx:13]: 700px assumption is out of sync with the rest of the Copilot app implementation.

File: src/app/detail/ui/Sidebar.tsx [hunk 4]
```diff
@@ -215,15 +215,15 @@ export const Sidebar = ({
...
+  const matches = useMediaQuery('(max-width:700px)')
 
   return (
     <Box
       sx={{
         borderLeft: (theme) => `1px solid ${theme.color.borders.border2}`,
         height: '100vh',
         display: 'block',
-        width: matches ? '100vw' : '25vw',
+        width: matches ? '100vw' : { sm: '16vw', md: '20vw' },
```
Comment (aatbip) [inline src/app/detail/ui/Sidebar.tsx:226]: I don't think we need to give varying height to solve this issue. Why don't we calculate the height of the title input box, and then calculate the height of the tapwrite description with respect to the height of the title input box? This makes the solution straightforward IMO.

**Review by aatbip — CHANGES_REQUESTED** [review]

File: src/app/detail/ui/Sidebar.tsx [hunk 5]
```diff
@@ -88,7 +88,7 @@ export const Sidebar = ({
         borderLeft: (theme) => `1px solid ${theme.color.borders.border2}`,
         height: '100vh',
         display: showSidebar ? 'block' : 'none',
-        width: matches && showSidebar ? '100vw' : '25vw',
+        width: matches && showSidebar ? '100vw' : { sm: '16vw', md: '20vw' },
```
Comment (aatbip) [inline src/app/detail/ui/Sidebar.tsx:91]: I think the width of the sidebar doesn't match the Figma design. According to the design, the sidebar occupied 29.5% of the total screen width which is equivalent to 366px, whereas in this PR the width of sidebar is very small i.e. ~ 336px.

File: src/app/detail/ui/ToggleController.tsx [hunk 6]
```diff
@@ -10,7 +10,7 @@ export const ToggleController = ({ children }: { children: ReactNode }) => {
   const { showSidebar } = useSelector(selectTaskDetails)
   const matches = useMediaQuery('(max-width:600px)')
 
-  const nonMobile = useMediaQuery('(min-width:600px)')
+  const nonMobile = useMediaQuery('(min-width:700px)')
```
Comment (aatbip) [inline src/app/detail/ui/ToggleController.tsx:13]: https://github.com/copilot-platforms/tasks-app/pull/267#discussion_r1735935876

**Review by aatbip — CHANGES_REQUESTED** [review]

File: src/app/detail/ui/Sidebar.tsx [hunk 7]
```diff
@@ -223,7 +223,7 @@ export const SidebarSkeleton = () => {
         borderLeft: (theme) => `1px solid ${theme.color.borders.border2}`,
         height: '100vh',
         display: 'block',
-        width: matches ? '100vw' : '25vw',
+        width: matches ? '100vw' : '366px',
       }}
```
Comment (aatbip) [inline src/app/detail/ui/Sidebar.tsx:227]: I think its better to use `calc` function here. Otherwise, `vw` should be nice.

**Review by aatbip — CHANGES_REQUESTED** [review]
For the screen sizes below 600px, we are getting an empty space in the bottom. 

Screenshot from 2024-08-30 17-39-35

**Pull request closed without merging**